lib/microkube/payload.rb

```diff
@@ -0,0 +1,48 @@
+require 'jwt'
+
+class PayloadError < StandardError
+end
+
+module Microkube
+  class Payload
+    def initialize(params)
+      @jwt_secret = params[:secret]
+      @jwt_exp = params[:expire]
+      raise PayloadError.new unless @jwt_secret
+      @jwt_exp = 600 unless @jwt_exp
+    end
+
+    def encode(token)
+      JWT.encode token, @jwt_secret, 'HS256'
+    end
+
+    def generate!(params)
+      raise PayloadError.new unless params[:service]
+      raise PayloadError.new unless params[:tag]
+
+      token = params.merge({
+        'iat': Time.now.to_i,
+        'exp': (Time.now + @jwt_exp).to_i
+      })
+
+      encode token
+    end
+
```

Comment:
name it verify!() if you don't need the other one don't implement it

Reply:
Should I add method to verify params before merge?